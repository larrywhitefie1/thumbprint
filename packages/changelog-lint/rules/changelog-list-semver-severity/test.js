const test = require('../../test-utils');
const rule = require('./');

const valid = [
    '* [Major] ',
    '* [Minor] ',
    '* [Patch] ',
    '* [Major] Foo',
    '* [Minor] Foo',
    '* [Patch] Foo',
];

const invalid = [
    '* Major ',
    '* Minor ',
    '* Patch ',
    '* [Majorr] ',
    '* [Minorr] ',
    '* [Patchh] ',
    '* [Major]',
    '* [Minor]',
    '* [Patch]',
];

test({
    rule,
    valid,
    invalid,
});
