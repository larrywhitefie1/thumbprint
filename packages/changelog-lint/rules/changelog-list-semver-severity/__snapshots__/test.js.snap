// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "* Major ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "* Minor ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #2 fails 1`] = `
Object {
  "input": "* Patch ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #3 fails 1`] = `
Object {
  "input": "* [Majorr] ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #4 fails 1`] = `
Object {
  "input": "* [Minorr] ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #5 fails 1`] = `
Object {
  "input": "* [Patchh] ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #6 fails 1`] = `
Object {
  "input": "* [Major]",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #7 fails 1`] = `
Object {
  "input": "* [Minor]",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #8 fails 1`] = `
Object {
  "input": "* [Patch]",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "List items must start with \`[Major]\`, \`[Minor]\`, or \`[Patch]\`. There must be a space after the severity.",
      "ruleId": "changelog-list-semver-severity",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
